salt/minion.py

```diff
@@ -984,8 +991,14 @@ def _spawn_minions(self):
                                                 loaded_base_name='salt.loader.{0}'.format(s_opts['master']),
                                                 jid_queue=self.jid_queue,
                                                )
-            self.minions.append(minion)
-            self.io_loop.spawn_callback(self._connect_minion, minion)
+
+            self._connect_minion(minion)
+
+        if not self.minions:
```

Comment:
`_connect_minion` is async operation. If it will not done fast enough the list could be empty here. It's better to run this check with `io_loop.call_later`.